src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/SampleServiceImpl.java

```diff
@@ -456,9 +456,15 @@ private SampleSequencingObjectJoin addSequencingObjectToSample(Sample sample, Se
 	public Collection<Sample> getSamplesForAnalysisSubmission(AnalysisSubmission submission) {
 		Set<SequencingObject> objectsForAnalysisSubmission = sequencingObjectRepository
 				.findSequencingObjectsForAnalysisSubmission(submission);
-
-		Set<Sample> samples = objectsForAnalysisSubmission.stream()
-				.map(s -> ssoRepository.getSampleForSequencingObject(s).getSubject()).collect(Collectors.toSet());
+		Set<Sample> samples = null;
+		try {
+			samples = objectsForAnalysisSubmission.stream()
+					.map(s -> ssoRepository.getSampleForSequencingObject(s)
+							.getSubject())
+					.collect(Collectors.toSet());
+		} catch (NullPointerException e) {
+			return null;
```

Comment:
`return new AnalysisSistrResults(null, true, null);` The first null is the sampleName and it will be set to null if there are no samples for the analysis. The view handles this so that other details for the sistr results can still be filled out even if the samples are not found. I've changed it to log a message rather than returning null if there is a NullPointerException. Updated in 6b29dcd95568eb2f6722f1be5f1d4fd201a9043f

Reply:
Good thanks for the explanation.  Can that log statement be bumped up to a `warn`?  That sounds like a problem but since it's handled later down in the chain that's ok.

Reply:
Bumped up to a warn in 736e177d65ec5882ff7682aece14faa0f46b8ca6